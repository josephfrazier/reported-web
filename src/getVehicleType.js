import axios from 'axios';

// ported from https://github.com/jeffrono/Reported/blob/19b588171315a3093d53986f9fb995059f5084b4/v2/enrich_functions.rb#L325-L346
export default function getVehicleType({ licensePlate, licenseState }) {
  const url = `https://findbyplate.com/US/${licenseState}/${licensePlate}/`;

  console.time(url); // eslint-disable-line no-console
  return axios.get(url).then(({ data }) => {
    console.timeEnd(url); // eslint-disable-line no-console
    const vehicleSummary = data
      .match(/<h2 class="vehicle-modal">(.+?)</s)[1]
      .trim();
    const components = vehicleSummary.split(' ');
    const vehicleYear = components[0];
    let vehicleMake = components[1];
    vehicleMake =
      vehicleMake.charAt(0).toUpperCase() + vehicleMake.slice(1).toLowerCase();
    const vehicleModel = components.slice(2).join(' ');
    let vehicleBody;

    try {
      vehicleBody = data
        .match(/<div class="cell" data-title="BodyClass">(.+?)</s)[1]
        .trim();
    } catch (err) {
      console.error('no vehicle body');
    }

    // if (vehicleYear === 'Try Members Area') {
    //   const message = 'not found';
    //   throw { message, licensePlate, licenseState }; // eslint-disable-line no-throw-literal
    // }

    return {
      result: {
        vehicleYear,
        vehicleMake,
        vehicleModel,
        vehicleBody,
        licensePlate,
        licenseState,
      },
    };
  });
}
