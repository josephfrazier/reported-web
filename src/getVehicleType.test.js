/**
 * @jest-environment node
 */
/* eslint-env jest */

import getVehicleType from './getVehicleType.js';

describe('getVehicleType', () => {
  test('returns the right object', async () => {
    const result = await getVehicleType({
      licensePlate: 'T716540C',
      licenseState: 'NY',
    });

    expect(result).toMatchSnapshot();
  });
});
