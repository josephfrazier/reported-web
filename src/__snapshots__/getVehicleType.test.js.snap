// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getVehicleType returns the right object 1`] = `
Object {
  "result": Object {
    "licensePlate": "T716540C",
    "licenseState": "NY",
    "vehicleBody": "Sedan/Saloon",
    "vehicleMake": "Lincoln",
    "vehicleModel": "Town Car",
    "vehicleYear": "2009",
  },
}
`;
